tests/random_attack.py

```diff
@@ -851,6 +864,12 @@ def join_self(self):
         self.dedup_names()
         return True
 
+    # This is a noop operation for the python data
+    def shallow_copy(self):
+        self.df_shallow_copy = self.df.copy()
+        self.df_deep_copy = copy.deepcopy(self.df)
```

Comment:
Does `copy.deepcopy` even works on a Frame?

Reply:
Looks like both `copy.copy()` and `copy.deepcopy()` invoke Frame's pickling protocol... We should probably adjust this behavior: https://docs.python.org/3/library/copy.html

Reply:
Yes, it works, I was like it is just doing pickling/unpickling, no? I've been using it in some FTRL tests and it seems to work fine.